app/models/king.rb

```diff
@@ -18,7 +18,7 @@ def valid_move?(x,y)
       [self.position_x-1,self.position_y],
       [self.position_x-1,self.position_y-1]
     ]
-    return valid_moves.include?([x,y]) && is_on_board?(x,y)
+    return valid_moves.include?([x,y]) && is_on_board?(x,y) && !is_in_check?(x,y)
```

Comment:
I have often found it more legible to break my return clauses into several steps;

```ruby
return false unless valid_moves.include?([x,y])
return false unless is_on_board?(x,y)
return !is_in_check?(x,y)
```

As a reader I can now scan and see specific conditions, and not have to comprehend the full chain of `&&`.

Reply:
Also, I believe adding `!is_in_check?(x,y)` is creating the infinite loop.